### PR TITLE
Break ajax stuff out out utils.cljs

### PR DESCRIPTION
**Pull request by MattPutnam**
Okay I'm done, I promise.

I also folded user_info.cljs into utils/user.cljs. Doing so introduced a circular dependency, so there's a new injection to fix that. I also ran across a TON of indentation that annoyed me, so I added custom indentations to the style.xml and fixed them all. Even with all of that, this PR is 99% moving stuff and formatting. The only nontrivial changes are that one injection, and I converted a function to a React component in links.cljs.

- [x] **Submitter**: Include the JIRA issue number in the PR description
- [x] **Submitter**: Check documentation and code comments. Add explanatory PR comments if helpful.
- [x] **Submitter**: If you changed a URL that is used elsewhere (e.g. in an email), comment about where it is used and ensure the dependent code is updated.
- [x] **Submitter**: JIRA ticket checks:
  * Acceptance criteria exists and is met
  * Note any changes to implementation from the description
  * To Demo flag is set
  * Release Summary is filled out, if applicable
  * Add notes on how to QA
- [x] **Submitter**: Update RC_XXX release ticket with any config or environment changes necessary
- [x] **Submitter**: Update FISMA documentation if changes to:
  * Authentication
  * Authorization
  * Encryption
  * Audit trails
- [x] **Submitter**: If you're adding new libraries, sign us up to security updates for them
- [x] **Submitter**: If you're adding new automated UI tests, review the test plan with QA
* Review cycle:
  * LR reviews
  * Rest of team may comment on PR at will
  * **LR assigns to submitter** for feedback fixes
  * Submitter rebases to develop again if necessary
  * Submitter makes further commits. DO NOT SQUASH
  * Submitter updates documentation as needed
  * Submitter **reassigns to LR** for further feedback
- [x] **TL** sign off
- [x] **LR** sign off
- [x] **Product Owner** sign off
- [x] **Submitter**: Verify all tests go green, including CI tests and automated UI tests.
- [x] **Submitter**: Squash commits and merge to develop. If adding test code, merge application code and test code at the same time.
- [x] **Submitter**: Delete branch after merge
- [x] **Submitter**: **Test this change works on dev environment after deployment**. YOU own getting it fixed if dev isn't working for ANY reason!
- [x] **Submitter**: Mark JIRA issue as resolved once this checklist is completed

### CODE REVIEW

File: src/cljs/main/broadfcui/common/links.cljs [hunk 1]
```diff
@@ -21,18 +21,17 @@
...
-     (fn [{:keys [props locals]}]
-       (let [{:keys [label object filename]} props
-             payload-blob (js/Blob. (js/Array. object) {:type "text/plain"})
-             payload-object-url (js/URL.createObjectURL payload-blob)]
-         (swap! locals assoc :objectUrl payload-object-url)
-         (create-download label payload-object-url filename)))
-     :component-will-unmount
-     (fn [{:keys [locals]}]
-       (js/URL.revokeObjectURL (:objectUrl @locals)))}) {:label label :object object :filename filename}])
+(react/defc DownloadFromObject
```
Comment (aednichols) [inline src/cljs/main/broadfcui/common/links.cljs:24]: I'm fine with either, but Isaac had suggested I do `create-download-from-object` so it's more obviously a sibling of `create-download`

File: src/cljs/main/broadfcui/auth.cljs [hunk 2]
```diff
@@ -9,6 +9,7 @@
    [broadfcui.config :as config]
    [broadfcui.nav :as nav]
    [broadfcui.utils :as utils]
```
Comment (aednichols) [inline src/cljs/main/broadfcui/auth.cljs:11]: It's a bit confusing that we still have a `utils` file that's a sibling to the `utils` directory, but maybe you have further plans for this

**Review by aednichols — APPROVED** [review]
👍 LGTM

File: src/cljs/main/broadfcui/auth.cljs (hunk 2 above)
Comment (MattPutnam) [reply src/cljs/main/broadfcui/auth.cljs:11]: It's the same kind of pattern that we have for table.

**Comment by davidangb** [comment]
@MattPutnam I removed myself as reviewer - as long as you get two thumbs, I'm good with this. If the PR is still open next week, I can review.

**Comment by MattPutnam** [comment]
jenkins retest

**Review by zarsky-broad — APPROVED** [review]
Didn't run.

**Pull request merged**